src/parsers/manifest/metaplaylist/metaplaylist_parser.ts

```diff
@@ -288,12 +295,10 @@ function createManifest(
     periods.push(...manifestPeriods);
 
     if (!isLive && duration != null) {
-      const currentDuration = currentManifest.getDuration();
-      if (currentDuration == null) {
-        duration = undefined;
-      } else {
-        duration += currentDuration;
+      if (lastEnd === null || firstStart === null) {
+        throw new Error("MPL Parser: can't define duration of manifest.");
```

Comment:
What? But you just checked that duration was defined!

Reply:
🐚